packages/changed-elements-react/src/widgets/V2Widget/componets/VersionCompareSelectModal.tsx

```diff
@@ -0,0 +1,359 @@
+        toastComparisonJobComplete({
+          comparisonJob: { comparisonJob: comparisonJob },
+          comparisonJobClient: args.comparisonJobClient,
+          iModelConnection: args.iModelConnection,
+          targetVersion: args.targetVersion,
+          currentVersion: args.currentVersion,
+        });
+      }
+      return;
+    }
+  }
+};
+
+type PostOrGetComparisonJobParams = {
+  changedElementsClient: IComparisonJobClient;
+  iTwinId: string;
+  iModelId: string;
+  startChangesetId: string;
+  endChangesetId: string;
+};
+
+async function postOrGetComparisonJob(args: PostOrGetComparisonJobParams): Promise<ComparisonJob> {
+  let result: ComparisonJob;
+  try {
+    result = await args.changedElementsClient.postComparisonJob({
+      iTwinId: args.iTwinId,
+      iModelId: args.iModelId,
+      startChangesetId: args.startChangesetId,
+      endChangesetId: args.endChangesetId,
+      headers: { "Content-Type": "application/json" },
+    });
+  } catch (error: unknown) {
+    if (error && typeof error === "object" && "code" in error && error.code !== "ComparisonExists") {
+      throw error;
+    }
+
+    result = await args.changedElementsClient.getComparisonJob({
+      iTwinId: args.iTwinId,
+      iModelId: args.iModelId,
+      jobId: `${args.startChangesetId}-${args.endChangesetId}`,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  return result;
+}
+
+type ManagerStartComparisonV2Args = {
+  comparisonJob: ComparisonJobCompleted;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  targetVersion: NamedVersion;
+  currentVersion: NamedVersion;
+};
+
+const runManagerStartComparisonV2 = async (args: ManagerStartComparisonV2Args) => {
+  if (VersionCompare.manager?.isComparing) {
+    return;
+  }
+  toastComparisonVisualizationStarting();
+  const changedElements = await args.comparisonJobClient.getComparisonJobResult(args.comparisonJob);
+  VersionCompare.manager?.startComparisonV2(args.iModelConnection, args.currentVersion, args.targetVersion, [changedElements.changedElements]).catch((e) => {
+    Logger.logError(VersionCompare.logCategory, "Could not start version comparison: " + e);
+  });
+};
+
+
+const toastComparisonJobProcessing = (currentVersion: NamedVersion, targetVersion: NamedVersion) => {
+  IModelApp.notifications.outputMessage(
+    new NotifyMessageDetails(
+      OutputMessagePriority.Info,
+      IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.versionPickerTitle"),
+      `${IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.iModelVersions")}
+      <${currentVersion?.displayName}> ${IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.and")} <${targetVersion.displayName}>
+      ${IModelApp.localization.getLocalizedString("VersionCompare:versionCompare.jobProcessing")}`,
+      OutputMessageType.Toast,
```

Comment:
There will be extra spaces in the message text after each line break. Push the message to the left while ignoring code indentation.